CompletionException every few seconds

(May be related to #2578, message seems similar but not identical)

I get this message every few seconds. Client is not attesting at the moment.
```
18:20:21.144 ERROR - PLEASE FIX OR REPORT | Unexpected exception thrown for p2p-async-scheduler-0 java.util.concurrent.CompletionException: java.util.concurrent.RejectedExecutionException: Task tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner$$Lambda$327/0x00007f69f9bc58b0@2f63d2c0 rejected from java.util.concurrent.ThreadPoolExecutor@3eba676c[Running, pool size = 10, active threads = 10, queued tasks = 5000, completed tasks = 1052560] (See log file for full stack trace)
```

Logfile:
```
2020-08-19 18:21:05.685+01:00 | P2PService-event-thread-0 | WARN  | Eth2TopicHandler | Encountered exception while processing message for topic /eth2/e7a75d5a/beacon_aggregate_and_proof/ssz_snappy
java.util.concurrent.CompletionException: java.util.concurrent.RejectedExecutionException: Task tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner$$Lambda$327/0x00007f69f9bc58b0@66c13d4 rejected from java.util.concurrent.ThreadPoolExecutor@3eba676c[Running, pool size = 10, active threads = 10, queued tasks = 5000, completed tasks = 1052560]
	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:367) [?:?]
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1108) [?:?]
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2235) [?:?]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.thenCompose(SafeFuture.java:373) [teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	at tech.pegasys.teku.networking.eth2.gossip.topics.Eth2TopicHandler.handleMessage(Eth2TopicHandler.java:43) [teku-networking-eth2-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	at tech.pegasys.teku.networking.p2p.libp2p.gossip.GossipHandler.apply(GossipHandler.java:75) [teku-networking-p2p-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	at tech.pegasys.teku.networking.p2p.libp2p.gossip.GossipHandler.apply(GossipHandler.java:33) [teku-networking-p2p-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	at io.libp2p.pubsub.PubsubApiImpl.onNewMessage(PubsubApiImpl.kt:79) [jvm-libp2p-minimal-0.5.5-RELEASE.jar:?]
	at io.libp2p.pubsub.PubsubApiImpl.access$onNewMessage(PubsubApiImpl.kt:22) [jvm-libp2p-minimal-0.5.5-RELEASE.jar:?]
	at io.libp2p.pubsub.PubsubApiImpl$1.invoke(PubsubApiImpl.kt:63) [jvm-libp2p-minimal-0.5.5-RELEASE.jar:?]
	at io.libp2p.pubsub.PubsubApiImpl$1.invoke(PubsubApiImpl.kt:22) [jvm-libp2p-minimal-0.5.5-RELEASE.jar:?]
	at io.libp2p.pubsub.AbstractRouter.onInbound(AbstractRouter.kt:212) [jvm-libp2p-minimal-0.5.5-RELEASE.jar:?]
	at io.libp2p.etc.util.P2PService.streamInbound(P2PService.kt:171) [jvm-libp2p-minimal-0.5.5-RELEASE.jar:?]
	at io.libp2p.etc.util.P2PService$StreamHandler$channelRead$1.invoke(P2PService.kt:51) [jvm-libp2p-minimal-0.5.5-RELEASE.jar:?]
	at io.libp2p.etc.util.P2PService$StreamHandler$channelRead$1.invoke(P2PService.kt:36) [jvm-libp2p-minimal-0.5.5-RELEASE.jar:?]
	at io.libp2p.etc.util.P2PService$runOnEventThread$1.run(P2PService.kt:231) [jvm-libp2p-minimal-0.5.5-RELEASE.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.util.concurrent.RejectedExecutionException: Task tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner$$Lambda$327/0x00007f69f9bc58b0@66c13d4 rejected from java.util.concurrent.ThreadPoolExecutor@3eba676c[Running, pool size = 10, active threads = 10, queued tasks = 5000, completed tasks = 1052560]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2055) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:825) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1355) ~[?:?]
	at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.runAsync(ScheduledExecutorAsyncRunner.java:68) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	at tech.pegasys.teku.networking.eth2.gossip.topics.Eth2TopicHandler.lambda$handleMessage$3(Eth2TopicHandler.java:45) ~[teku-networking-eth2-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1106) ~[?:?]
	... 20 more
2020-08-19 18:21:06.603+01:00 | p2p-async-scheduler-0 | ERROR | teku-status-log | PLEASE FIX OR REPORT | Unexpected exception thrown for p2p-async-scheduler-0
java.util.concurrent.CompletionException: java.util.concurrent.RejectedExecutionException: Task tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner$$Lambda$327/0x00007f69f9bc58b0@201adb2e rejected from java.util.concurrent.ThreadPoolExecutor@3eba676c[Running, pool size = 10, active threads = 10, queued tasks = 5000, completed tasks = 1052560]
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088) ~[?:?]
	at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.handleExecutorError(ScheduledExecutorAsyncRunner.java:113) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$runAfterDelay$0(ScheduledExecutorAsyncRunner.java:90) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.util.concurrent.RejectedExecutionException: Task tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner$$Lambda$327/0x00007f69f9bc58b0@201adb2e rejected from java.util.concurrent.ThreadPoolExecutor@3eba676c[Running, pool size = 10, active threads = 10, queued tasks = 5000, completed tasks = 1052560]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2055) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:825) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1355) ~[?:?]
	at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$runAfterDelay$0(ScheduledExecutorAsyncRunner.java:88) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	... 6 more
2020-08-19 18:21:06.956+01:00 | p2p-async-scheduler-0 | ERROR | teku-status-log | PLEASE FIX OR REPORT | Unexpected exception thrown for p2p-async-scheduler-0
java.util.concurrent.CompletionException: java.util.concurrent.RejectedExecutionException: Task tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner$$Lambda$327/0x00007f69f9bc58b0@5c2cb7c5 rejected from java.util.concurrent.ThreadPoolExecutor@3eba676c[Running, pool size = 10, active threads = 10, queued tasks = 5000, completed tasks = 1052560]
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088) ~[?:?]
	at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.handleExecutorError(ScheduledExecutorAsyncRunner.java:113) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$runAfterDelay$0(ScheduledExecutorAsyncRunner.java:90) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.util.concurrent.RejectedExecutionException: Task tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner$$Lambda$327/0x00007f69f9bc58b0@5c2cb7c5 rejected from java.util.concurrent.ThreadPoolExecutor@3eba676c[Running, pool size = 10, active threads = 10, queued tasks = 5000, completed tasks = 1052560]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2055) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:825) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1355) ~[?:?]
	at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$runAfterDelay$0(ScheduledExecutorAsyncRunner.java:88) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-1bbea6b0]
	... 6 more

```

--- a/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/Eth2TopicHandler.java
+++ b/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/Eth2TopicHandler.java
@@ -15,6 +15,7 @@ package tech.pegasys.teku.networking.eth2.gossip.topics;
 
 import com.google.common.base.Throwables;
 import io.libp2p.core.pubsub.ValidationResult;
+import java.util.concurrent.RejectedExecutionException;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes;
@@ -63,6 +64,9 @@ public abstract class Eth2TopicHandler<T extends SimpleOffsetSerializable, TWrap
   protected ValidationResult handleMessageProcessingError(Throwable err) {
     if (Throwables.getRootCause(err) instanceof DecodingException) {
       LOG.trace("Received malformed gossip message on {}", getTopic());
+    } else if (Throwables.getRootCause(err) instanceof RejectedExecutionException) {
+      LOG.warn(
+          "Discarding gossip message for topic {} because the executor queue is full", getTopic());
     } else {
       LOG.warn("Encountered exception while processing message for topic {}", getTopic(), err);
     }

--- a/teku/src/main/java/tech/pegasys/teku/TekuDefaultExceptionHandler.java
+++ b/teku/src/main/java/tech/pegasys/teku/TekuDefaultExceptionHandler.java
@@ -14,11 +14,13 @@
 package tech.pegasys.teku;
 
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
 import com.google.common.eventbus.SubscriberExceptionContext;
 import com.google.common.eventbus.SubscriberExceptionHandler;
 import java.lang.Thread.UncaughtExceptionHandler;
 import java.lang.reflect.Method;
 import java.nio.channels.ClosedChannelException;
+import java.util.concurrent.RejectedExecutionException;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import tech.pegasys.teku.events.ChannelExceptionHandler;
@@ -87,6 +89,9 @@ public final class TekuDefaultExceptionHandler
       LOG.debug("Channel unexpectedly closed", exception);
     } else if (isSpecFailure(exception)) {
       statusLog.specificationFailure(subscriberDescription, exception);
+    } else if (Throwables.getRootCause(exception) instanceof RejectedExecutionException) {
+      LOG.error(
+          "Unexpected rejected execution due to full task queue in {}", subscriberDescription);
     } else {
       statusLog.unexpectedFailure(subscriberDescription, exception);
     }

